tempo_embeddings/embeddings/weaviate_database.py

```diff
@@ -0,0 +1,278 @@
+    ):
+        super().__init__(batch_size)
+        self.db_path = db_path
+        self.embedder_name = embedder_name
+        self.embedder_config = embedder_config or {"type": "default"}
+        self.tokenizer = AutoTokenizer.from_pretrained(embedder_name) if embedder_name else None
+        self.model = None
+        self.client = None
+        self.weaviate_headers = {}
+        
+        if self.embedder_config.get("type") == "hf":
+            self.embedding_function = wvc.config.Configure.Vectorizer.text2vec_huggingface(model=self.embedder_name)
+            self.weaviate_headers = {"X-HuggingFace-Api-Key": self.embedder_config["api_key"]}
+        elif self.embedder_config.get("type") == "custom_model" or self.embedder_config.get("type") == "default":
+            try:
+                self.model = self.embedder_config["model"]
+                self.model.batch_size = self.batch_size
+                self.embedding_function = wvc.config.Configure.Vectorizer.none()
+            except KeyError:
+                logger.error("If the type is 'custom_model' or 'default' you should pass the model object under Key 'model'")
+        else:
+            raise ValueError(f"Malformed embedder_config {self.embedder_config}. Check that 'type', 'api_key' and 'model' keys are properly populated.")
+
+        if os.path.exists(self.db_path):
+            self._load_config()
+        else:
+            os.makedirs(self.db_path)
+            self.config = {
+                "embedder_name": self.embedder_name,
+                "embedder_type": self.embedder_config["type"],
+                "existing_collections": [],
+            }
+        
+        
+    def connect(self):
+        with weaviate.connect_to_local() as client:
+            # Do something with the client
+            logger.info(type(client))
+            logger.info("Weaviate Server Is Up: %s", client.is_ready())
+            return client.is_ready()
+
+
+    def _save_config(self):
+        if self.config:
+            with open(f"{self.db_path}/config.json", "w", encoding="utf-8") as f:
+                json.dump(self.config, f, indent=2)
+    
+
+    def _load_config(self):
+        with open(f"{self.db_path}/config.json", encoding="utf-8") as f:
+            self.config = json.load(f)
+
+
+    def get_available_collections(self):
+        if self.config:
+            return self.config.get("existing_collections", [])
+        return []
+
+
+    def create_new_collection(
+        self,
+        name: str,
+        corpus: Corpus = None,
+        # collection_metadata = None, # This Dict can be used in the future to give a precise SCHEMA. NOT IMPLEMENTED YET!
+    ) -> None:
+
+        # Create NEW collection and Embeds the given passages. Do nothing otherwise
+        if name in self.config["existing_collections"]:
+            raise ValueError("This collection has been created already! Try 'insert_corpus()' if you want to add more items to the collection")
```

Comment:
I don't want to be pedantic, but adding the collection name might help debugging:
```suggestion
            raise ValueError(f"Collection '{name}' has been created already! Try 'insert_corpus()' if you want to add more items to the collection")
```